src/graph/topo.cc

```diff
@@ -607,6 +607,65 @@ ncclResult_t ncclTopoGetLocalNet(struct ncclTopoSystem* system, int rank, int64_
   return ncclSuccess;
 }
 
+ncclResult_t scklGetTopoFromXMLAndSetChannels(struct ncclComm* comm) {
+  char* str = getenv("SCKL_XML_FILE");
+  if (str){
+    INFO(NCCL_ENV, "SCKL_XML_FILE set by environment to %s", str);
+    struct ncclXml* xml;
+    NCCLCHECK(ncclCalloc(&xml, 1));
+    NCCLCHECK(scklTopoGetXmlGraphFromFile(str, xml));
```

Comment:
Not clear to me that you need a separate sckl specific XML parser. Can you make NcclTopoGetXmlGraphFromFile to work here?